sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/VariableLengthRowBasedKeyValueBatch.java

```diff
@@ -141,14 +145,16 @@ public boolean next() {
           return false;
         }
 
-        totalLength = Platform.getInt(base, offsetInPage) - 4;
-        currentklen = Platform.getInt(base, offsetInPage + 4);
+        int uaoSize = UnsafeAlignedOffset.getUaoSize();
+        int doubleUaoSize = uaoSize + uaoSize;
```

Comment:
let's be consistent with https://github.com/apache/spark/pull/28195/files#diff-ed82c6d7b012a3bf89fc2974e2840deaR46